### DISCUSSION TITLE
Channel1 event

### DISCUSSION THREAD

**Discussion by BusinessClaes** [arendst/Tasmota#16825] [Q&A] [answered]
Hi,
I’m trying to fetch the event for change in Channel1 and store its value in a variable. Doing this for a Dimmer event is no problem, just listening to ”on event dimmer1#state do var1 %value%” does the trick.

But doing this with Channel1 seems to be more complicated. Channel1#state doesn’t work, and then I think am close to getting it to work with Channel1#data but I don’t get it all the way. See below, the response to “Channel1” command works but “Channel1 50” doesn’t work since the response has more values than just Channel1. Any idea about how to do this?

`CMD: Channel1 50 `
`MQT: stat/TMT_RGBW03/RESULT = {"POWER1":"ON","Channel1":50,"Color":"80"} `
(nothing happens)

`CMD: Channel1 `
`MQT: stat/TMT_RGBW03/RESULT = {"Channel1":50} `
`RUL: CHANNEL1#DATA performs "var2 50"`
`MQT: stat/TMT_RGBW03/RESULT = {"Var2":"50"} `
(this is what I need to happen)

The rule I’m using is 
`on channel1#data do var2 %value% endon `

**Comment by barbudor** [comment, marked as answer]
Works on my side without the `#data`

```
12:05:44.960 CMD: channel1
12:05:44.972 MQT: stat/esp32dev3/CHANNEL = {"Channel1":50}
12:05:44.985 RUL: CHANNEL1 performs "var2 "
12:05:44.996 MQT: stat/esp32dev3/VAR = {"Var2":"50"}
12:05:48.948 CMD: channel1 50
12:05:48.959 MQT: stat/esp32dev3/CHANNEL = {"POWER1":"ON","Channel1":50,"Color":"80"}
12:05:48.973 RUL: CHANNEL1 performs "var2 50"
12:05:48.984 MQT: stat/esp32dev3/VAR = {"Var2":"50"}

12:06:46.071 MQT: stat/esp32dev3/RULE = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":33,"Free":478,"Rules":"on channel1 do var2 %value% endon"}}
``

**Reply by BusinessClaes** [reply]
Thanks a lot! That solved it!

**Reply by sfromis** [reply]
Note that if you want to pick up Channel1 from both types of payloads (with or without other fields present), you need two triggers, both with and without `#data`. Single-value payloads in response to a command are treated as a special case.

**Reply by barbudor** [reply]
That what I thought but here the trigger without data works in both cases

**Reply by BusinessClaes** [reply]
> That what I thought but here the trigger without data works in both cases

Well, the trigger without "#data" fires the rule in both cases, but the value is only fetched with "channel1#data" for single-value payload and with plain "channel1" for multiple-value payload. So depending on the case, one or the other trigger should be used as @sfromis points out.

Thank you guys for the help, really appreciated!
